tests/integration/components/spread-test.js

```diff
@@ -1,30 +1,235 @@
+        options: {
+          property: 'Neat',
+          onClick: handler
+        }
+      })
+
+      this.render(hbs`
+        {{spread-test
+          options=options
+        }}
+      `)
+    })
+
+    it('should bind spread properties as local properties', function () {
+      expect($hook('spread-property').text().trim()).to.equal('Neat')
+    })
+
+    it('should bind spread functions as local functions', function () {
+      $hook('spread-test').click()
+
+      expect(handler).to.have.callCount(1)
+    })
+
+    it('should update local properties when the source property changes', function () {
+      this.set('options.property', 'Woah')
+
+      expect($hook('spread-property').text().trim()).to.equal('Woah')
+    })
+  })
+
+  describe('when using a custom spread property', function () {
+    beforeEach(function () {
+      this.setProperties({
+        options: {
+          property: 'Neat',
+          onClick: handler
+        }
+      })
+
+      this.render(hbs`
+        {{spread-test
+          foo=options
+          spreadOptions=(hash
+            property='foo'
+          )
+        }}
+      `)
+    })
+
+    it('should bind spread properties as local properties', function () {
+      expect($hook('spread-property').text().trim()).to.equal('Neat')
+    })
+
+    it('should bind spread functions as local functions', function () {
+      $hook('spread-test').click()
+
+      expect(handler).to.have.callCount(1)
+    })
+
+    it('should update local properties when the source property changes', function () {
+      this.set('options.property', 'Woah')
+
+      expect($hook('spread-property').text().trim()).to.equal('Woah')
+    })
+  })
+
+  describe('when providing a source binding', function () {
+    describe('and using the default (options) spread property', function () {
+      beforeEach(function () {
+        this.set('options', Ember.Object.create({}))
+
+        this.render(hbs`
+          {{spread-test
+            options=options
+            spreadOptions=(hash
+              source=(hash
+                object=this
+                property='options'
+              )
+            )
+          }}
+        `)
+      })
+
+      it('should bind new source properties as local properties', function () {
+        this.set('options.property', 'Crazy')
+
+        expect($hook('spread-property').text().trim()).to.equal('Crazy')
+      })
+
+      it('should bind new source functions as local functions', function () {
+        this.set('options.onClick', handler)
+
+        $hook('spread-test').click()
+
+        expect(handler).to.have.callCount(1)
+      })
+
+      it('should update new local properties when the source property changes', function () {
+        this.set('options.property', 'Crazy')
+
+        expect($hook('spread-property').text().trim()).to.equal('Crazy')
+
+        this.set('options.property', 'Booya')
+
+        expect($hook('spread-property').text().trim()).to.equal('Booya')
+      })
```

Comment:
In this `it()` you've duplicated some of what you've tested in the `it()` two above this. Instead of having that independently written twice, I'd use a nested `describe`

```js
describe('when a new source property is added', function () {
  beforeEach(function () {
    this.set('options.property', 'Crazy')
  })

  it('should bind it as a local property', function () {
    expect($hook('spread-property').text().trim()).to.equal('Crazy')
  })

  describe('when the source property changes', function () {
    beforeEach(function () {
      this.set('options.property', 'Booya')
    })
    
    it('should update the local property', function () {
      expect($hook('spread-property').text().trim()).to.equal('Booya')
    }) 
  })
})
```